hubblestack/hec/obj.py

```diff
@@ -0,0 +1,327 @@
+
+import socket
+import json
+import time
+import copy
+import os
+import hashlib
+
+import certifi
+import urllib3
+
+from . dq import DiskQueue
+
+# fakelogging goes to a file iff specified in environment
+# and otherwise does nothing (for debugging)
+import fakelogging as logging
+log = logging.getLogger('obj')
+
+__version__ = '1.0'
+
+_max_content_bytes = 100000
+http_event_collector_debug = False
+
+# the list of collector URLs given to the HEC object
+# are hashed into an md5 string that identifies the URL set
+# these maximums are per URL set, not for the entire disk cache
+max_diskqueue_items = 200
```

Comment:
This is likely too small. We probably just want to control with disk size, and put this to "unlimited" or an arbitrarily high number.

Reply:
I feel like the items limiter is usually going to be better. The disk size is somewhat random and I feel like (without being able to justify it fully) it's going to be more reliable. 200 may be a bit low. I really have no idea what reasonable settings for either of these are though, so we'll 0 this one and set the size to 50megs or something? 100megs? It's per hec endpoint, so probably a bit lower is better.

Reply:
Yeah I'm not sure either. I expect @fossam will want it "as high as possible" but have no concept of how many events we'll need to buffer to actually cover small outages.